<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1167</width>
    <height>627</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTableWidget" name="fieldTable">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>551</width>
      <height>441</height>
     </rect>
    </property>
    <property name="dragEnabled">
     <bool>false</bool>
    </property>
    <property name="dragDropMode">
     <enum>QAbstractItemView::NoDragDrop</enum>
    </property>
    <property name="defaultDropAction">
     <enum>Qt::IgnoreAction</enum>
    </property>
    <property name="alternatingRowColors">
     <bool>true</bool>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>true</bool>
    </attribute>
    <attribute name="horizontalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>35</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>20</number>
    </attribute>
    <column>
     <property name="text">
      <string>Item</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Type</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Endianness</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Base</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Length</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Index</string>
     </property>
    </column>
   </widget>
   <widget class="QTextEdit" name="dissectorEdit">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>90</y>
      <width>531</width>
      <height>441</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>551</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <widget class="QTextEdit" name="fieldNameBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="addButton">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>50</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>ADD</string>
     </property>
    </widget>
    <widget class="QComboBox" name="baseBox">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>50</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>DEC</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>HEX</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>OCT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>NONE</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="typeBox">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>BOOLEAN</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>CHAR</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UINT8</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UINT16</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UINT32</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UINT64</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>INT8</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>INT16</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>INT32</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>INT64</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>FLOAT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>DOUBLE</string>
      </property>
     </item>
    </widget>
    <widget class="QCheckBox" name="endianCheckbox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Little Endian</string>
     </property>
    </widget>
    <widget class="QPushButton" name="deleteButton">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>50</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>DELETE ROW</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>30</y>
      <width>531</width>
      <height>51</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QPushButton" name="generateButton">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>10</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>GENERATE</string>
     </property>
    </widget>
    <widget class="QLabel" name="acronymLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Acronym</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="portEdit">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>10</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>10000</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="nameEdit">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>10</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>mydissector</string>
     </property>
    </widget>
    <widget class="QLabel" name="nameLabel">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>10</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Name</string>
     </property>
    </widget>
    <widget class="QLabel" name="portLabel">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Port</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="acronymEdit">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>10</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>41</width>
       <height>31</height>
      </size>
     </property>
     <property name="placeholderText">
      <string>ABC</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="saveAsButton">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>540</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>SAVE AS...</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1167</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
